api/internal/plugins/execplugin/execplugin.go

```diff
@@ -93,7 +93,9 @@ func (p *ExecPlugin) processOptionalArgsFields() error {
 		return err
 	}
 	if c.ArgsOneLiner != "" {
-		p.args, _ = shlex.Split(c.ArgsOneLiner)
+		p.args, _ = shell.Fields(c.ArgsOneLiner, func(name string) string {
```

Comment:
> This seems to do a lot more than what we want--we only want to split the string using shell rules, but `shell.Fields` is focussed on shell expansion, which we do not want--ideally the string doesn't contain anything to be expanded, but if it does, substitution should happen during the real invocation rather than here.

It would be a risk.

> I'm struggling to find a replacement as well. I wonder if we should vendor the code, if we cannot keep the archived dep.

if the code will not be changed, https://github.com/kubernetes/utils/pull/232 may be a workaround to maintain the code in `github.com/kubernetes/utils`.

Reply:
Maintaining it in kubernetes/utils sounds reasonable to me.